src/VirtoCommerce.SimpleExportImportModule.Data/Services/CsvPagedPriceDataSource.cs

```diff
@@ -83,30 +84,45 @@ public async Task<bool> FetchAsync()
                 return false;
             }
 
-            // CSV Reader can only move forward, i.e. Skip will not work: after reading N items we can read only next N items
-            var records = _csvReader.GetRecords<CsvPrice>().Take(PageSize).ToArray();
-            CurrentPageNumber++;
+            var recordTuples = new List<(CsvPrice, string, int)>();
+
+            for (var i = 0; i < PageSize && _csvReader.Read(); i++)
```

Comment:
There is async csv helper read method. Why we don't use it?

Reply:
switched to ReadAsync